Automattic/wp-calypso
# Plans data-store: refactor unit tests and fix TypeScript errors

## Changes proposed in this Pull Request

* Add mock data for tests that mimics real-world data
* Fix TypeScript errors
* Change a few types based on the data coming from API endpoints (including making `path_slug` optional). Types from APIs data have been simplified (only properties that are actually used are typed, in order to make it more explicit)

## Testing instructions

- `yarn test-packages:watch packages/data-stores/src/plans` — the plans data-store tests should pass
- Smoke test plans grid on Gutenboarding, Step by Step launch and Focused Launch
  - For good measure, open all plans grid ts/tsx files and check that there are not TypeScript errors

Closes #49536
